### ISSUE TITLE
SSL Support for RMIRegistryExploit

### ISSUE THREAD

**Issue by TorgeH** [frohoff/ysoserial#33]
Hi,

trying to PoC-Exploit a setup like the one explained here:
https://blogs.oracle.com/jmxetc/entry/jmx_connecting_through_firewalls_using
(ssl enhanced RMI over single port), I'm stumbling upon the same error message as the first commenter.
Would it be possible to add ssl-support to RMIRegistryExploit?

**Comment by frohoff** [comment]
The exploit code would need to create the registry client with something like this, and probably customize the `SSLContext` to trust the cert(s) in question:

```java
SSLContext sc = SSLContext.getInstance("TLS");
// probably need to customize trust for self-signed certs here
SSLSocketFactory sf = sc.getSocketFactory();
RMIClientSocketFactory csf = new MySslRMIClientSocketFactory(sf);
Registry registry = LocateRegistry.getRegistry("myserver", 1234, csf);
```

It should be easier to build-in support for this when we finish #23.

**Comment by TorgeH** [comment]
Thanks for your observations and pointers.
I will try to tinker with these a bit.

**Comment by frohoff** [comment]
Renamed and tagged as enhancement

**Comment by frohoff** [comment]
This will probably have to wait for #23.

**Comment by frohoff** [comment]
This was fixed in #81
